src/ebonite/core/objects/wrapper.py

```diff
@@ -350,6 +350,14 @@ def __init__(self, model, *args, **kwargs):
         self.model = model
         self.refs = {}
 
+        # we couldn't import hook and analyzer at top as it leads to circular import failure
+        from ebonite.core.analyzer.model import CallableMethodModelHook, ModelAnalyzer
+        known_types = set()
+        for hook in ModelAnalyzer.hooks:
+            if not isinstance(hook, CallableMethodModelHook):
+                known_types.update(hook.valid_types)
```

Comment:
here you assume that `ModelHook` always has `valid_types` field, but it is not clear from it's defenition. I suggest add this field to `ModelHook` and maybe get rid of `TypeHookMixin` entirely, or just leave it with method implementations

Reply:
We couldn't get rid of `TypeHookMixin` as callable hook and even tf 1.x hook don't subclass it. I propose to leave code as it is but make this part more clear in documentation (we already provide some recommendations here on implementing support for new libs). What do you think?

Reply:
ok, leave it/ but problem is that if someone will implement hook, dont subclass TypeHookMixin and will not provide `valid_types` class field, this code will fail. Actually, logic of this relies on the fact that every hook implementation besides CallableMethodModelHook has meaningful `valid_types` field. So I propose move it to ModelHook class (and maybe even check that it is filled on subclass declaration)

Reply:
fixed